tensorflow/lite/delegates/nnapi/nnapi_delegate.cc

```diff
@@ -5133,16 +5134,26 @@ TfLiteStatus StatefulNnApiDelegate::DoPrepare(TfLiteContext* context,
   // Check for every node if it is supported
   const bool is_accelerator_specified = ShouldUseTargetDevices(
       delegate_options, nnapi, /*exclude_nnapi_reference=*/true);
+  std::vector<delegate::nnapi::NNAPIValidationFailure> map_failures;
   for (int node_index : TfLiteIntArrayView(plan)) {
     TfLiteNode* node;
     TfLiteRegistration* registration;
     TF_LITE_ENSURE_STATUS(context->GetNodeAndRegistration(
         context, node_index, &node, &registration));
     if (NNAPIDelegateKernel::Validate(context, registration->builtin_code,
                                       registration->version, target_sdk_version,
-                                      node, is_accelerator_specified)) {
+                                      node, is_accelerator_specified,
+                                      &map_failures)) {
       supported_nodes.push_back(node_index);
     }
+    for (auto& failure : map_failures) {
```

Comment:
I would surround all these additional lines with #ifdef NNAPI_VERBOSE_VALIDATION for consistency.

Reply:
Agree, thanks for the comment. The additional code is guarded with #ifdef.